lib/login.rb

```diff
@@ -21,12 +21,17 @@ def new_token_by_password(user:, password:, env:)
       debug_res = Http.request(**opts)
 
       # try challenging client auth
-      res = oauth_bearer_token_challenge(
-        server_url: env.api_endpoint_url,
-        user: user,
-        password: password,
-        proxy: env.client_proxy
-      )
+      res = {}
+      (1..3).each { |seconds|
+        res = oauth_bearer_token_challenge(
+          server_url: env.api_endpoint_url,
+          user: user,
+          password: password,
+          proxy: env.client_proxy
+        )
+        break if res[:success]
+        sleep (seconds * 30)
```

Comment:
Thanks for having this! Why do we increase the sleep seconds by each iteration?

Reply:
+1 for this question, rest looks good to me, thanks !!

Reply:
@jhou1 @kasturinarra  The idea is that we wait longer for another retry, as previous retry failed means the cluster haven't recovered itself.